[pigeon] fix named parameters in flutter api

schultek opened the pull request:
Fixes a bug where incorrect code is generated for methods with named parameters.

Fixes https://github.com/flutter/flutter/issues/140045

- Correctly generates code for flutter api methods with named parameters
- Correctly generates test code for host api methods with named parameters

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [ ] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

schultek commented on packages/pigeon/lib/ast.dart:
```diff
@@ -277,7 +277,7 @@ class Parameter extends NamedType {
     bool? isPositional,
     bool? isRequired,
     super.documentationComments,
-  })  : isNamed = isNamed ?? true,
```
This fixes existing tests that don't specify this but still assume it's false.

Since this wasn't used anywhere besides toString, I don't consider this a breaking change.

tarrinneal commented:
Can you add an integration test for this as well? If you need guidance on this, let me know.

 This will also need versioning (changelog, pubspec, and generator_tools.dart)

schultek commented:
@tarrinneal Thanks for reviewing so far.

I updated the versioning stuff and started with the integration tests, but I need your help on this:

I added some appropriate methods here: [packages/pigeon/pigeons/core_tests.dart](https://github.com/flutter/packages/pull/5663/files#diff-6efc8ba082c93b93a94d9fb4552aac409826896fdbfc54f941c03557e8188c01)
But generating and updating all the resulting native files is a lot and I neither have the knowledge nor tooling installed to handle all the platforms. Can you help by generating the files and updating the native implementations? Then I can continue with the dart parts.

tarrinneal commented:
There's actually another bug that's surfaced from this (I actually intended for this feature to only be on hostApi's). I'm going to patch into this pr, unless you prefer I don't.

tarrinneal commented:
closing in favor of https://github.com/flutter/packages/pull/5689

@schultek Thank you for putting this pr together. Unfortunately for now, we are going to be removing this feature as it was never intended to be added, and needs more work than just adding these fixes.
